### ISSUE TITLE
Gulp.watch: How to wait until previous task is not finished

### ISSUE THREAD

**Issue by webzepter** [gulpjs/gulp#1682]
Hello. I have an issue with `gulp.watch`.

**TLDR**: 
How to tell `gulp.watch` to wait until necessary tasks will not be finished before running new cycle?

**Reasons**:
I have three tasks: `clean`, `build`, and `watch`.
Task `clean` deletes directory, task `build` produces files in the same directory (`clean` is dependency for `build`).
Task `watch` runs `gulp.watch` on directory and then it runs `build` (that runs `clean`).

But sometimes `watch` task runs new `build` before previous `build` is finished and an error occurs: `build` continues write to directory and `clean` tries to remove directory (of course, `ENOTEMPTY` error happens).

**Comment by phated** [comment]
Support questions should be directed to StackOverflow.
